src/main/java/org/tomitribe/auth/signatures/SigningAlgorithm.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.tomitribe.auth.signatures;
+
+import java.util.Map;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+
+/**
+ * The algorithm parameter contains the name of the signature's Algorithm,
+ * as registered in the HTTP Signature Algorithms Registry defined by this document.
+ * 
+ * The signature verification is based on the signature's algorithm from the keyId
+ * parameter rather than from this algorithm.
+ * If algorithm is provided and differs from or is incompatible with the algorithm
+ * or key material identified by keyId (for example, algorithm has a value of
+ * rsa-sha256 but keyId identifies an EdDSA key), then a verification exception is
+ * raised.
+ * 
+ * The default value for this parameter should be "hs2019".
+ * 
+ * @see https://www.ietf.org/id/draft-ietf-httpbis-message-signatures-00.html
+ */
+public enum SigningAlgorithm {
+
+    /**
+     * The actual cryptographic algorithm is derived from metadata associated
+     * with keyId.
+     * 
+     * Recommend support for:
+     *   RSASSA-PSS [RFC8017] using SHA-512 [RFC6234]
+     *   HMAC [RFC2104] using SHA-512 [RFC6234]
+     *   ECDSA using curve P-256 [DSS] and SHA-512 [RFC6234]
+     *   Ed25519ph, Ed25519ctx, and Ed25519 [RFC8032]
+     */
+    HS2019("hs2019", null),
+
+    // Deprecated, SHA-1 is not secure.
```

Comment:
Do you think we should mark these with `@Deprecated`?

Reply:
Sure, that makes sense.